### PR TITLE
Major refactor to controllers and DRY improvement

### PR DESCRIPTION
**Pull request by CaiqueMitsuoka**

### CODE REVIEW

File: app/controllers/playlists_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class PlaylistsController < ApplicationController
+  def index
+    @playlists = current_user.playlists
+  end
+
+  def select
+    session[:current_playlist] = params.permit(:playlist)[:playlist]
```
Comment (CarolPC) [inline app/controllers/playlists_controller.rb]: Is it possible to replace the params.permit with params.require?

Comment (CarolPC) [inline app/controllers/playlists_controller.rb]: Maybe you could replace the params.permit with params.require?

**Comment by castanhob** [comment]
:top:

**Comment by castanhob** [comment]
:top: !

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,3 +1,12 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  private
+  def current_user
+    @user || RSpotify::User.new(session[:user_hash])
```
Comment (guivinicius) [inline app/controllers/application_controller.rb]: Use memoization here `@user ||=`

Also i think you should use session here to store user related info

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+  end
+
+  def current_playlist
+    @playlist || RSpotify::Playlist.find(current_user.id , session[:current_playlist])
```
Comment (guivinicius) [inline app/controllers/application_controller.rb]: Memoization

File: app/controllers/playlists_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+class PlaylistsController < ApplicationController
+  def index
+    @playlists = current_user.playlists
+  end
+
+  def select
```
Comment (guivinicius) [inline app/controllers/playlists_controller.rb]: Not sure if select is a good action name what is the purpose of that action ??

**Pull request merged**